src/hooks/use-chat-context.tsx

```diff
@@ -95,42 +96,33 @@ export const ChatProvider: React.FC< ChatProviderProps > = ( { children } ) => {
 	}, [] );
 
 	useEffect( () => {
-		let isCurrent = true;
 		const run = async () => {
 			const siteId = selectedSite?.id;
 			if ( ! siteId || selectedSite.isAddingSite ) {
 				return;
 			}
-			setInitialLoad( ( prev ) => ( { ...prev, [ siteId ]: true } ) );
-			try {
-				const result = await Promise.all( [ fetchPluginList( siteId ), fetchThemeList( siteId ) ] );
-				if ( isCurrent ) {
+			initialLoad.current[ siteId ] = true;
+			Promise.all( [ fetchPluginList( siteId ), fetchThemeList( siteId ) ] )
+				.then( ( result ) => {
 					setPluginsList( ( prev ) => ( { ...prev, [ siteId ]: result[ 0 ] } ) );
 					setThemesList( ( prev ) => ( { ...prev, [ siteId ]: result[ 1 ] } ) );
-				}
-			} catch ( error ) {
-				if ( isCurrent ) {
-					setInitialLoad( ( prev ) => ( { ...prev, [ siteId ]: false } ) );
-				}
-			}
+				} )
+				.catch( ( _error ) => {
+					initialLoad.current[ siteId ] = false;
+				} );
 		};
 		if (
 			selectedSite &&
 			! loadingSites &&
-			! initialLoad[ selectedSite.id ] &&
-			isCurrent &&
+			! initialLoad.current[ selectedSite.id ] &&
```

Comment:
Now changes to `initialLoad` won't trigger this side effect. This is needed to avoid an infinite loop if WP-CLI commands fail when fetching themes or plugins.